test_task_1.py

```diff
@@ -0,0 +1,28 @@
+# Our goal is to create a fullname field by concatenating a first and last name
+import unittest
+import task_1
+import random
+
+
+class TestFullNameCreated(unittest.TestCase):
+    def setUp(self) -> None:
+        self.user_records = task_1.create_full_name()
+        self.record_to_test = random.choice(self.user_records)  #avoid wasting time testing every record
+        self.split_name = self.record_to_test["full_name"].split()
```

Comment:
Ideally, tests should have the minimum implementation code necessary to assert the expected behaviour. If we start adding implementation in the tests, we can end up with implementation bugs in the tests (which we are not testing).

Reply:
I think @stortiz-lifeworks makes a good point here about using split.

Another way to test this could be to create a new test fixture file, something like `"./assets/user-with-full-name.json"` and then check that when your code runs it creates matching records. It looks like there's something like 10 test records - so it's just about doable. Then you get all of your type checks etc for free and you don't need to solve this problem of splitting the name apart within the test itself.

Not saying you need to do it this way btw - just sharing an idea on an alternative approach :)

Reply:
This is exactly the kind of feedback I was hoping for! I'll keep in mind not adding more-than-necessary implementation into future tests. For now I've removed the first and last name tests-- thinking about it, that functionality is tested pretty well by the regular expression check (and that doesn't require any extra implementation)